async-nats/src/jetstream/kv/mod.rs

```diff
@@ -644,11 +644,15 @@ impl Store {
     ///     })
     ///     .await?;
     /// kv.put("key", "value".into()).await?;
-    /// kv.delete("key").await?;
+    /// kv.delete("key", None).await?;
     /// # Ok(())
     /// # }
     /// ```
-    pub async fn delete<T: AsRef<str>>(&self, key: T) -> Result<(), DeleteError> {
+    pub async fn delete<T: AsRef<str>>(
+        &self,
+        key: T,
+        revison: Option<u64>,
```

Comment:
I don't think it's a good idea to add expected revision as a second parameter.
Firstly, it adds a mandatory parameter that is not often used, secondly, it's unnecesarry breaking change, but most importantly - it can be confusing, as someone who does not know what is the purpose of that revision might think that it is a revision to be deleted.

I suggest a different approach, like we do with publish_with_headers:

`delete_if_revision()` or
`purge_expect_revision()`

as additional methods.
Builder pattern feels like an overkill for just one option.

Reply:
I have adjusted the PR. The mandatory parameter is removed in favor of the added *_expect_revision functions.